### PR TITLE
Fix #550 - Use dlang.org theme on the editor page

### LINKED ISSUE #550: Use dlang.org theme on the editor page
Currently the editor page is pretty bare-bones:

https://tour.dlang.org/editor

Would be nice to use the dlang.org theme, at least for the top navigation bar.

### PR DESCRIPTION
**Pull request by wilzbach**
Instead of just dumping the menu, I went the "long" road and build the menu statically in D. 
This should allow us more flexibility in the future.

### CODE REVIEW

File: public/static/css/menu.css [hunk 1]
```diff
@@ -141,6 +141,15 @@
...
 
+#top #cssmenu > ul > li .menu-divider {
+    border-top: 1px solid #943029;
+}
+
+.expand-container > .expand-toggle::after {
+    content: " \f0d7";
+    font-family: FontAwesome;
+}
+
```
Comment (wilzbach) [inline public/static/css/menu.css]: Backported from dlang.org

File: public/static/js/menu.js [hunk 2]
```diff
@@ -14,6 +14,10 @@ if (typeof cssmenu_no_js === 'undefined') {
         if (!isHamburger) {
             open_main_item = container.classList.contains('open') ? container: null;
         }
+
+		if (container.className.indexOf("expand-container")) {
+			return;
+		}
```
Comment (wilzbach) [inline public/static/js/menu.js]: this is required, s.t. the link clicks on "Downloads" and "Packages" (i.e. no sections) don't get silently swallowed.

**Comment by wilzbach** [comment]
A few screenshots:

image

image

### Mobile

image

image

**Comment by CyberShadow** [comment]
> I went the "long" road and build the menu statically in D

That's a good idea only if you want to maintain a third copy of the menu...

I'd be inclined to say that this is a better idea:
https://github.com/dlang/dlang.org/blob/master/forum-template.dd

**Comment by wilzbach** [comment]
> That's a good idea only if you want to maintain a third copy of the menu...

I think there are more copies out there: blog.dlang.org, code.dlang.org, VisualD, ...

> I'd be inclined to say that this is a better idea:

Maybe in theory, but we already have adopted and trimmed down the menu CSS to a [single CSS file](https://github.com/dlang-tour/core/blob/master/public/static/css/menu.css) and we need to maintain the ability to generate the menu dynamically anyways for the language content.
Also except for the menu CSS there's nothing taken over from dlang.org.

**Comment by CyberShadow** [comment]
> I think there are more copies out there: blog.dlang.org, code.dlang.org, VisualD, ...

More things to fix, then.

The next time a major website redesign comes about, we are going to be neck deep into doing pointless work updating all these websites that could have been avoided with a bit of fore-planning and refactoring.

Except forum.dlang.org, of course.

> Maybe in theory, but we already have adopted and trimmed down the menu CSS to a single CSS file and we need to maintain the ability to generate the menu dynamically anyways for the language content.

Instead of working around things in an unmaintainable manner on your side, please refactor things on the dlang.org side to maximize reuse.

File: source/webinterface.d [hunk 3]
```diff
@@ -166,13 +166,87 @@ class WebInterface
 				toc, title, githubRepo, translations)();
 	}
 
+	private static auto buildDlangToc()
```
Comment (ZombineDev) [inline source/webinterface.d]: Looks so much better than all those DDoc shenanigans at dlang.org!

**Comment by wilzbach** [comment]
> The next time a major website redesign comes about, we are going to be neck deep into doing pointless work updating all these websites that could have been avoided with a bit of fore-planning and refactoring.

Btw it's much more likely that we redesign the tour in the meanwhile, e.g. https://github.com/dlang-tour/core/pull/473 ;-)

**Comment by ZombineDev** [comment]
> Instead of working around things in an unmaintainable manner on your side, please refactor things on the dlang.org side to maximize reuse.

The sooner we move dlang.org to vibe.d diet templates, the sooner this problem would be resolved :P

**Comment by wilzbach** [comment]
> The sooner we move dlang.org to vibe.d diet templates, the sooner this problem would be resolved :P

FWIW https://github.com/dlang/dlang.org/blob/master/dpl-docs/views/layout.dt

**Comment by wilzbach** [comment]
@CyberShadow I think it would be an unreasonable amount of work to import the the dlang.org tour Ddoc templete. Also we can't use submodules here, because sadly DUB still doesn't support them and the tour can be run offline (well currently this only works if you have the language translation, but it's an existing feature).

Lastly as mentioned before we/I plan to redesign the DTour a bit in the next few weeks, so we most likely are just going to keep the color schemes and style of dlang.org, which is only CSS anyhow.

**Pull request merged**